# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/IPCIdentityCredential.ipdlh
# Commit: 459739221a88
# Full Hash: 459739221a884b2b1c2adab42fd19fafe279bd20
# Author: Sandor Molnar <smolnar@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing build bustages
#   
#   This reverts commit ab681e36aeafc21ba62bb4680bdaa6206f02546d.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r fc9986230203 -r 459739221a88 dom/credentialmanagement/identity/IPCIdentityCredential.ipdlh
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/credentialmanagement/identity/IPCIdentityCredential.ipdlh	Thu Jun 05 00:19:41 2025 +0300
@@ -0,0 +1,31 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include "mozilla/dom/PermissionMessageUtils.h";
+[RefCounted] using class nsIPrincipal from "nsIPrincipal.h";
+
+namespace mozilla {
+namespace dom {
+
+// Most of these fields are taken directly from mozilla::dom::IdentityCredentialInit
+// Others are:
+//    identityProvider: the principal that created the IdentityCredential this represents.
+//    infoExpiresAt: PR_Now() + the mozilla::dom::IdentityCredentialInit's infoExpiresAfter, in ms
+struct IPCIdentityCredential
+{
+  nsString id;
+  nsString? token;
+  nsCString? name;
+  nsCString? iconURL;
+  nsCString[] effectiveOrigins;
+  nsCString? effectiveQueryURL;
+  nsCString? effectiveType;
+  uint64_t? infoExpiresAt;
+  nullable nsIPrincipal identityProvider;
+};
+
+}
+}